src/libraries/Common/src/Interop/OSX/Interop.libc.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libc
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct AttrList
+        {
+            [MarshalAs(UnmanagedType.U2)] public ushort bitmapCount;
+            [MarshalAs(UnmanagedType.U2)] public ushort reserved;
+            [MarshalAs(UnmanagedType.U4)] public uint commonAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint volAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint dirAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint fileAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint forkAttr;
```

Comment:
You don't need any of those `[MarshalAs]` attributes. The fields will be blitted properly since `uint` and `ushort` are all blittable primitives.